client/src/components/body/profilepage/components/GroupForm.js

```diff
@@ -94,49 +129,115 @@ const GroupForm = (props) => {
+            placeholder="Enter Group Name Here"
           />
         </FormGroup>
-        <FormGroup check>
-          <Label for="assign-lab">Choose labs to assign</Label>
+        <hr />
+        <FormGroup check className={"tw-flex tw-flex-col tw-w-full"}>
+          <div
+            className={
+              "tw-w-full tw-flex tw-flex-col tw-justify-between tw-text-left"
+            }
+          >
+            <p className={"tw-title-styling-name tw-font-poppins tw-text-lg"}>
+              Assign Color for Group:
+            </p>
+            <p className={"tw-font-calibri tw-font-medium"}>
+              {" "}
+              Select a color for this group to distinguish it from the others!{" "}
+            </p>
+          </div>
+          <div className={"tw-grid tw-grid-cols-2 tw-gap-5 tw-w-1/2 tw-py-5"}>
+            {cardColors.map((color, key) => {
+              return (
+                <div
+                  key={key}
+                  className={
+                    "tw-flex tw-flex-row tw-h-[5rem] tw-w-[10rem] tw-items-center"
+                  }
+                >
+                  <Input
+                    type="radio"
+                    name="groupColor"
+                    className={"tw-m-3 tw-rounded-sm"}
+                    defaultChecked={
+                      groupColor ? groupColor.includes(color) : false
+                    }
+                    onChange={() => {
+                      setColor(`group_${color}`);
+                      console.warn("Group color changed");
+                    }}
+                  />
+                  <div
+                    className={`group_${color} tw-h-full tw-w-full tw-rounded-t-lg tw-p-3`}
+                  />
+                </div>
+              );
+            })}
+          </div>
+        </FormGroup>
+        <hr />
+        <h4 className={"tw-font-poppins tw-title-styling-name tw-text-lg"}>
+          {" "}
+          Assign Labs:{" "}
+        </h4>
+        <FormGroup check className={"tw-px-5"}>
+          <p>Select all labs you wish to assign to this group.</p>
           {labs.map((lab) => (
-            <div key={lab.id}>
+            <div
+              key={lab.id}
+              className={
+                "tw-flex tw-flex-row tw-items-center tw-gap-3 tw-pl-[5%]"
+              }
+            >
               <Input
+                className={"tw-rounded-sm tw-box-content"}
                 type="checkbox"
                 name={lab.id}
                 id={"lab" + lab.id}
                 checked={!!checkedLabs[lab.id]}
                 onChange={() => toggleCheck(lab.id)}
               />
-              <Label for={"lab" + lab.id}>{lab.labShortName}</Label>
+              <div
+                className={`tw-w-full tw-my-3 ${checkedLabs[lab.id] && "tw-border-solid tw-border-primary-blue tw-rounded-xl"}`}
+              >
+                <LabRow lab={lab} />
+              </div>
             </div>
```

Comment:
![Screenshot 2024-10-25 at 1 13 55 PM](https://github.com/user-attachments/assets/db574e98-3fd5-4cd4-afa3-f412f009f8a2)
Clicking the "More Information" button doesn't close the previous modal. Not a priority but this should be fixed

Reply:
There's also no way to close the "More Information" modal? That should be fixed